Source/WebGPU/WebGPU/UsdModelRenderer.swift

```diff
@@ -0,0 +1,428 @@
+import Foundation
+import Metal
+import QuartzCore
+import QuartzCore_Private
+import WebGPU_Internal
+#if canImport(DirectDrawBackend)
+import DirectDrawBackend
+import DirectResource
+import ShaderGraph
+#endif
+
+@objc
+@implementation
+extension DDBridgeVertexAttributeFormat {
+    let semantic: Int32
+    let format: Int32
+    let layoutIndex: Int32
+    let offset: Int32
+
+    init(
+        semantic: Int32,
+        format: Int32,
+        layoutIndex: Int32,
+        offset: Int32
+    ) {
+        self.semantic = semantic
+        self.format = format
+        self.layoutIndex = layoutIndex
+        self.offset = offset
+        super.init()
+    }
+}
+
+@objc
+@implementation
+extension DDBridgeVertexLayout {
+    let bufferIndex: Int32
+    let bufferOffset: Int32
+    let bufferStride: Int32
+
+    init(
+        bufferIndex: Int32,
+        bufferOffset: Int32,
+        bufferStride: Int32
+    ) {
+        self.bufferIndex = bufferIndex
+        self.bufferOffset = bufferOffset
+        self.bufferStride = bufferStride
+        super.init()
+    }
+}
+
+@objc
+@implementation
+extension DDBridgeAddMeshRequest {
+    var indexCapacity: Int32
+    var indexType: Int32
+    var vertexBufferCount: Int32
+    var vertexCapacity: Int32
+    var vertexAttributes: [DDBridgeVertexAttributeFormat]?
+    var vertexLayouts: [DDBridgeVertexLayout]?
+
+    init(value: Int32) {
+        self.indexCapacity = 0
+        self.indexType = 0
+        self.vertexBufferCount = 0
+        self.vertexCapacity = 0
+        super.init()
+    }
+    #if canImport(DirectDrawBackend)
+    fileprivate func makeDescriptor() -> DirectResource.MeshDescriptor {
```

Comment:
```suggestion
    fileprivate var descriptor: DirectResource.MeshDescriptor {
```

rule of thumb: generally only use functions when I/O is being performed or side effects happen, otherwise properties are usually preferred